tests/debugpy/test_evaluate.py

```diff
@@ -677,3 +677,88 @@ def code_to_debug():
 
         session.request_continue()
         assert backchannel.receive() == 1000
+
+
+def test_evaluate_thread_locks(pyfile, target, run):
+    @pyfile
+    def code_to_debug():
+        """
+        The idea here is that a secondary thread does the processing of instructions,
+        so, when all threads are stopped, doing an evaluation for:
+
+        processor.process('xxx')
+
+        would be locked until secondary threads start running.
+        See: https://github.com/microsoft/debugpy/issues/157
+        """
+
+        import threading
```

Comment:
Needs to do:
```py
import debuggee
debuggee.setup()
```
to work with all the runners (`attach_pid`, `attach_listen`, and `attach_connect` all rely on this). It's is the standard scaaffolding for debuggee code, and should only be skipped when the test is doing some kind of highly custom launch or attach sequence all by itself (i.e. without using a runner).